lens/lily/modules/statemanager.go

```diff
@@ -22,7 +22,7 @@ var log = logging.Logger("lily/lens")
 var ErrExecutionTraceNotFound = fmt.Errorf("failed to find execution trace")
 
 func StateManager(lmctx helpers.MetricsCtx, lc fx.Lifecycle, cs *store.ChainStore, exec stmgr.Executor, sys vm.SyscallBuilder, us stmgr.UpgradeSchedule, bs beacon.Schedule, em stmgr.ExecMonitor) (*stmgr.StateManager, error) {
-	sm, err := stmgr.NewStateManagerWithUpgradeScheduleAndMonitor(cs, exec, sys, us, bs, em)
+	sm, err := stmgr.NewStateManagerWithUpgradeScheduleAndMonitor(cs, exec, sys, us, bs, em, nil)
```

Comment:
what is the `nil` value being passed to the `stmgr`? I assume it's a datastore? The statemanager will likely need a concrete value passed here to operate.

Reply:
Got it. But I am not familiar with it here. Could you help me to do it here?  thanks!

Reply:
Here is the constructor from lotus: https://github.com/filecoin-project/lotus/blob/master/node/modules/stmgr.go#L13

Reply:
So this should look something like:

```go
func StateManager(lmctx helpers.MetricsCtx, lc fx.Lifecycle, cs *store.ChainStore, exec stmgr.Executor, sys vm.SyscallBuilder, us stmgr.UpgradeSchedule, bs beacon.Schedule, em stmgr.ExecMonitor, metadataDs dtypes.MetadataDS) (*stmgr.StateManager, error) {
	sm, err := stmgr.NewStateManagerWithUpgradeScheduleAndMonitor(cs, exec, sys, us, bs, em, metadataDs)
```

Reply:
This is the package lotus and lily use for dep injection like this: https://github.com/uber-go/fx

Reply:
Fixed! Thanks for your help!